src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -388,128 +385,54 @@ private SNITCPHandle CreateTcpHandle(DataSource details, long timerExpire, objec
             }
 
             int port = -1;
-            if (details.IsSsrpRequired)
+            if (details.ConnectionProtocol == DataSource.Protocol.Admin)
             {
-                try
+                if (details.IsSsrpRequired)
```

Comment:
There are two branches leading to the similar code.
Can this be changed to do something like

```C#

if(details.IsSsrpRequired)
{
    try 
        {
               port = details.ConnectionProtocol == DataSource.Protocol.Admin ? 
                                   SSRP.GetDacPortByInstanceName(hostName, details.InstanceName) :
                                   SSRP.GetPortByInstanceName(hostName, details.InstanceName);
       }
       catch(SocketException se)
       {
              ...
       }
}
```
And then in the last else block 
```
{
   port = details.ConnectionProtocol == DataSource.Protocol.Admin ? 
                                  DefaultSqlServerDacPort : DefaultSqlServerPort;
}
```